docs/pages/ESSupport.md

```diff
@@ -0,0 +1,368 @@
+
+## ECMAScript Version Support
+
+The UI5 Tooling offers general support for the latest ECMAScript features. While a `ui5 build` is executed, the UI5 Tooling analyses the code in a few places. Depending on your project you have to consider some restrictions regarding the used ECMAScript syntax.
+
+## General
+The following sections describe the limitations valid for all types of projects.
+
+### sap.ui.define/sap.ui.require: Usage of Object Expressions, Spread Elements, and Template Literals
+An `Object Expression`, `Spread Element`, or `Template Literal` with an expression is not supported as a parameter for `sap.ui.define` or `sap.ui.require`.
+
+**✅ Best Practice:**
+```javascript
+sap.ui.define(["Bar"], function(Bar){
+});
+```
+
+**⛔️ Not supported: Object Expression**
+```javascript
+const dependency = "Bar";
+sap.ui.define([dependency], function(Bar){
+});
+```
+
+**⛔️ Not supported: Spread Element**
+```javascript
+const dependenciesA = ["ModuleA1", "ModuleA2"];
+const dependenciesB = ["ModuleB1", "ModuleB2"];
+sap.ui.define([...dependenciesA, ...dependenciesB],
+	function(ModuleA1, ModuleA2, ModuleB1, ModuleB2){
+});
+```
+
+**⛔️ Not supported: Template Literal with Expression**
+```javascript
+sap.ui.define([`ModuleA`, `ModuleB` ], () => {
```

Comment:
I find this example confusing. All places except the `Module${i}` are supported, aren't they?

Reply:
const dependenciesA = ["ModuleA", "ModuleB"];
const i = `D`;
return sap.ui.require([
			`ModuleC`,
			`Module${i}`
		], () => { });

Reply:
done